### DISCUSSION TITLE
Cooler digest with 2 restriction enzymes

### DISCUSSION THREAD

**Discussion by ddepierre** [open2c/cooler#400] [Q&A] [answered]
Hi cooler people, 

Is it possible to use cooler digest with 2 restriction enzymes?

Something like:
`
cooler digest --rel-ids 0 --out mm10_DpnII_DdeI_restriction_sites_cooler_digest.bed chrNameLength.txt mm10.chroms.all.fa DdeI, DpnII`

or an alternative with cooltools genome digest like:
`cooltools genome digest chrNameLength.txt mm10.chroms.all.fa DdeI,DpnII`

Thanks for support, 
David

**Comment by ddepierre** [comment]
Alternatively, can you confirm I can do it with 
```

mm10_DdeI_restriction_sites_cooler_digest_3cols.bed = rtracklayer::import.bed('mm10_DdeI_restriction_sites_cooler_digest_3cols.bed')
mm10_MboI_restriction_sites_cooler_digest_3cols.bed = rtracklayer::import.bed('mm10_MboI_restriction_sites_cooler_digest_3cols.bed')
mm10_DdeI_MboI_restriction_sites_cooler_digest_3cols.bed = GenomicRanges::disjoin(c(mm10_DdeI_restriction_sites_cooler_digest_3cols.bed, mm10_MboI_restriction_sites_cooler_digest_3cols.bed))
rtracklayer::export.bed(mm10_DdeI_MboI_restriction_sites_cooler_digest_3cols.bed, 'mm10_DdeI_MboI_restriction_sites_cooler_digest_3cols.bed')


```
Giving this, combining the blue and the green in the red one:

![image](https://github.com/open2c/cooler/assets/25797403/67b7dde9-f4d4-4efd-82dd-0be7d788700b)


Best, 
David

**Comment by nvictus** [comment, marked as answer]
`cooler digest` does not support multiple enzymes, though this is something we can add, since it uses biopython under the hood <s>which can do this</s>. EDIT: on closer inspection, I'm not sure it can. 

>Alternatively, can you confirm I can do it with

If I understand correctly what GRanges `disjoin` is doing (cutting out overlapping ranges as separate intervals), then, yes, this makes sense!
